src/odemis/gui/cont/acquisition.py

```diff
@@ -2272,50 +2274,126 @@ def _on_ac_done(self, future):
 
 
 class FastEMAlignmentController:
-    def __init__(self, tab_data, panel):
+    """
+    Takes care of the calibration button and process in the calibration panel in the FastEM acquisition tab.
+    """
+    def __init__(self, tab_data, tab_panel):
+        """
+        tab_data: (FastEMAcquisitionGUIData) The representation of the microscope GUI.
+        tab_panel: (wx.Frame) The calibration panel, which contains the calibration button,
+                    the gauge and the label of the gauge.
+        """
         self._tab_data = tab_data
-        self._panel = panel
+        self._main_data_model = tab_data.main
+        self._panel = tab_panel
+
+        self.btn_align = tab_panel.btn_align
+        self.gauge_progress = tab_panel.align_gauge_progress  # progress bar
+        self.gauge_label = tab_panel.align_lbl_gauge  # progress bar label
+        self.is_aligned = tab_data.main.is_aligned
+
+        self.gauge_progress.Hide()  # hide progress bar
+        self._panel.Parent.Layout()
+        self.btn_align.Bind(wx.EVT_BUTTON, self.on_align)
+
+        # FIXME: How to differentiate between simulator and real HW?
+        #   The following two calibrations can also be run with the simulator.
+        #   However, the rest not. How make compartible with both cases?
+        # CALIBRATIONS_SIM and CALIBRATIONS as global var somehow?
+        self.calibrations = [OPTICAL_AUTOFOCUS, IMAGE_TRANSLATION_PREALIGN]
+
+        # check calibration state of system
+        # If backend was not restarted, but only GUI, then the system is in principle still calibrated.
+        # However, so far when closing the GUI the system is also not calibrated anymore.
+        # self.is_aligned.subscribe(self._on_align_state, init=True) # TODO needed?
 
-        self._panel.align_gauge_progress.Hide()
-        panel.btn_align.Bind(wx.EVT_BUTTON, self._on_btn_align)
-        tab_data.main.is_aligned.subscribe(self._on_align_state, init=True)
+        if not self.is_aligned.value:
+            self._reset_calibration_gui()  # display estimated calibration time
```

Comment:
I'm not sure I understand what this does. Does it reset the estimated calibration time?

Reply:
By default there are some random 30 seconds displayed, which I don't know where they come from. So I wanted here to actually set the calibration time needed. If we manage that the system is still calibrated after closing the GUI, we could have an else statement which resets the gui to show the label "calibrated".

Reply:
ok